File: 287-find-the-duplicate-number/287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int n = nums.length;
        
        for(int i=0; i<n; i++) {
            int idx = Math.abs(nums[i])-1;
            if(nums[idx] > 0) nums[idx] = -nums[idx];
            else return Math.abs(nums[i]);
        }
        
        return -1;
    }
}

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
//     int findDuplicate(vector<int>& nums) {
//         int n = nums.size();
        
//         for(int i=0; i<n; i++) {
//             int idx = nums[i]-1;
//             if(nums[idx] < 0) return nums[i];
            
//             nums[idx] *= -1;
//         }
        
//         return -1;
//     }
    int findDuplicate(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        
        int last = -1;
        for(int i: nums) {
            if(i==last) return i;
            last = i;
        } 
        return -1;
    }
};